File: app/src/main/java/ru/p3tr0vich/fuel/BroadcastReceiverLoading.java
package ru.p3tr0vich.fuel;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v4.content.LocalBroadcastManager;

abstract class BroadcastReceiverLoading extends BroadcastReceiverLocalBase {

    private static final String ACTION = BuildConfig.APPLICATION_ID + ".ACTION_LOADING";
    private static final String EXTRA_LOADING = BuildConfig.APPLICATION_ID + ".EXTRA_LOADING";

    @Override
    protected final String getAction() {
        return ACTION;
    }

    public static void send(@NonNull Context context,
                            boolean loading) {
        LocalBroadcastManager.getInstance(context).sendBroadcast(new Intent(ACTION)
                .putExtra(EXTRA_LOADING, loading));
    }

    @Override
    public final void onReceive(Context context, Intent intent) {
        onReceive(intent.getBooleanExtra(EXTRA_LOADING, false));
    }

    public abstract void onReceive(boolean loading);
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/receivers/BroadcastReceiverLoading.kt
package ru.p3tr0vich.fuel.receivers

import android.content.Context
import android.content.Intent
import android.support.v4.content.LocalBroadcastManager
import ru.p3tr0vich.fuel.BuildConfig

abstract class BroadcastReceiverLoading : BroadcastReceiverLocalBase() {

    override val action: String
        get() = ACTION

    override fun onReceive(context: Context, intent: Intent) {
        onReceive(intent.getBooleanExtra(EXTRA_LOADING, false))
    }

    abstract fun onReceive(loading: Boolean)

    companion object {
        private const val ACTION = "${BuildConfig.APPLICATION_ID}.ACTION_LOADING"
        private const val EXTRA_LOADING = "${BuildConfig.APPLICATION_ID}.EXTRA_LOADING"

        @JvmStatic
        fun send(context: Context, loading: Boolean) {
            LocalBroadcastManager.getInstance(context)
                    .sendBroadcast(Intent(ACTION).putExtra(EXTRA_LOADING, loading))
        }
    }
}